### ISSUE TITLE
Generic.__new__ should pass args/kwargs to super-class __new__

### ISSUE THREAD

**Issue by chadrik** [python/typing#514]
Is there a reason that `Generic.__new__`  does not pass `*args` and `**kwargs` to the super class's `__new__`?  These are required in some cases for proper behavior.  

Here's a somewhat non-sensical example that demonstrates the problem:

```python
from __future__ import absolute_import, print_function

from typing import Generic, TypeVar
T = TypeVar('T')

class BaseThing(Generic[T]):
    def __new__(cls, *args, **kwargs):
        kwargs.setdefault('encoding', 'utf-8')
        return super(BaseThing, cls).__new__(cls, *args, **kwargs)

class Str(BaseThing, unicode):
    pass

s = Str('blarg')
print(repr(s))
```

`s` is `u''` instead of `u'blarg'`.

The offending code is in the [`typing._generic_new()`](https://github.com/python/typing/blob/master/python2/typing.py#L1286) helper.

**Comment by ilevkivskyi** [comment]
This may be just an oversight (since we pass them to ``__init__``). Would you like to try to make a PR?
(Be sure to add tests if yes.)

**Comment by chadrik** [comment]
PR here: #517

**Comment by ilevkivskyi** [comment]
I think this was fixed by #517.
